### ISSUE TITLE
 Gutenberg: v10.0.x plugin upgrade

### ISSUE THREAD

**Issue by fullofcaffeine** [Automattic/wp-calypso#50060]
Previous upgrade: https://github.com/Automattic/wp-calypso/issues/49534
Release notes: [v10.0.0](https://github.com/WordPress/gutenberg/releases/tag/v10.0.0), [v10.0.1](https://github.com/WordPress/gutenberg/releases/tag/v10.0.1), [v10.0.2](https://github.com/WordPress/gutenberg/releases/tag/v10.0.2)

### Process

- [x] Install and activate
  - [x] Copy & install v10.0.0-rc.1: D57096-code, D57097-code
  - [x] Activate v10.0.0-rc.1 on edge: D57114-code
  - [x] Copy & install v10.0.0: D57206-code, D57208-code
  - [x] Activate v10.0.0 on edge: D57209-code
  - [x] Activate v10.0.0 on production: D57115-code
  - [x] Request update for Atomic sites: p9o2xV-1gZ-p2
  - [x] Copy & install v10.0.1: D57444-code, D57445-code
  - [x] Activate v10.0.1 on production: D57448-code
  - [x] Copy & install v10.0.2: D57511-code, D57512-code
  - [x] Activate v10.0.2 on edge: D57513-code
  - [x] Activate v10.0.2 on production: D57514-code
- [x] Testing
  - [x] Calypso E2E:
    - [x] v10.0.0: https://github.com/Automattic/wp-calypso/pull/50108 ~(currently blocked by D57303-code)~
    - [x] v10.0.1: https://github.com/Automattic/wp-calypso/pull/50343
    - [x] v10.0.2: https://github.com/Automattic/wp-calypso/pull/50387
  - [x] Team City:
    - [x] v10.0.0: D57115-code
    - [x] v10.0.1: D57448-code
    - [x] v10.0.2: D57514-code
- [x] Publish internal announcements: p7DVsv-aAS-p2, p1613994381011300-slack-CB0B2G43X
- [x] Clean up unused releases
  - [x] v9.8.4: D57418-code
  - [x] v10.0.0-rc.1: D57212-code
  - [x] v10.0.0: D57665-code
  - [x] v10.0.1: D57673-code
- [x] Update the Gutenberg Upgrade Log [pcoGjb-g-p2] and list patches applied: pcoGjb-W-p2
- [x] Open a new issue for the next upgrade, transfer remaining tasks, close this issue.

### Issues / Blockers 🐛 

- [x] The `core/block-editor` events tracking is broken because the store namespace is not always a string now.
  - Fixed in https://github.com/Automattic/wp-calypso/pull/50243 / D57303-code

- [x] G2 component's CSS parsing error causing WSOD
  - Reported in: p1613999993009200-slack-CBTN58FTJ
  - Upstream issue: https://github.com/ItsJonQ/g2/issues/266
  - Upstream fix: https://github.com/ItsJonQ/g2/pull/267
  - WPCOM hotfix: D57423-code
  - **Actual** upstream fix: https://github.com/WordPress/gutenberg/issues/29248 (p1614035275052000-slack-C7YPUHBB2)
  - Gutenberg patch release: https://github.com/WordPress/gutenberg/releases/tag/v10.0.2
---
FYI: @Automattic/team-calypso

**Comment by WunderBart** [comment]
This one's done! 🎉 Next one: https://github.com/Automattic/wp-calypso/issues/50323

**Comment by WunderBart** [comment]
Reopened because the bug reported in https://github.com/ItsJonQ/g2/issues/266 is causing WSOD for some users and we need a patch release. See more details in this issue's description.

**Comment by WunderBart** [comment]
Reopening for the [v10.0.2 patch](https://github.com/WordPress/gutenberg/releases/tag/v10.0.2) upgrade.
